app/models/user_link.rb

```diff
@@ -1,3 +1,14 @@
 class UserLink < ActiveRecord::Base
+  before_save :fix_link
   belongs_to :user
+  scope :with_link, ->{ where('link IS NOT NULL') }
+
+  def link_without_protocol
```

Comment:
As the class is called UserLink, I would rename this method to without_protocol to avoid repetition.